src/bridge/cockpitpeer.c

```diff
@@ -265,9 +290,33 @@ on_other_control (CockpitTransport *transport,
 
           if (!self->last_init)
             {
-              default_init = g_strdup_printf ("{ \"command\": \"init\", \"version\": 1, \"host\": \"%s\" }",
-                                       self->init_host ? self->init_host : "localhost");
-              self->last_init = g_bytes_new_take (default_init, strlen (default_init));
+              JsonObject *object = cockpit_transport_build_json ("command", "init", NULL);
+              json_object_set_int_member (object, "version", 1);
+              json_object_set_string_member (object, "host", self->init_host ? self->init_host : "localhost");
+
+              if (explicit_superuser_capability)
+                {
+                  const gchar *superuser = "any";
+                  if (self->init_superuser && *self->init_superuser)
+                    superuser = self->init_superuser;
+
+                  if (!g_str_equal (superuser, "none"))
+                    {
+                      JsonObject *superuser_options;
+
+                      superuser_options = json_object_new ();
+                      json_object_set_string_member (superuser_options, "id", superuser);
+                      json_object_set_object_member (object, "superuser", superuser_options);
+                    }
+                  else
+                    {
+                      json_object_set_boolean_member (object, "superuser", FALSE);
+                      cockpit_peer_delete_authorize_values (self);
```

Comment:
This will also set `self->authorize_values_timeout` to zero, without actually removing the timeout. That's wrong and might explain the segfault we have seen occasionally in CI.